### PR TITLE
chore(build): Refactor to simplify and add dedicated test mode

### PR DESCRIPTION
**Pull request by rhuss**
Rearranged script a bit to make easier to understand the various run modes.
Added a `--test` mode which will only run the tests

### CODE REVIEW

File: hack/build.sh [hunk 1]
```diff
@@ -267,10 +281,12 @@ ln -s $(basedir)/hack/build.sh /usr/local/bin/kn_build.sh
 
 Examples:
 
-* Compile, format, tests, docs:  build.sh
-* Compile only:                  build.sh --fast
-* Compile with tests:            build.sh -f -t
-* Automatice recompilation:      build.sh --watch
+* Update deps, format, license check,
+  gen docs, compile, test: .,,,....... build.sh
```
Comment (navidshaikh) [inline hack/build.sh]: s/gen/generate
s/,/.

File: hack/build.sh [hunk 2]
```diff
@@ -55,38 +55,52 @@ run() {
...
+    if $(has_flag --test -t); then
+       go_test
+    fi
+    exit 0
+  fi
 
-    # Auto generate cli docs
-    generate_docs
+  # Run only tests
+  if $(has_flag --test -t); then
```
Comment (navidshaikh) [inline hack/build.sh]: thanks!
this was needed

**Comment by rhuss** [comment]
@maximilien @sixolet could we get this build optimization reviewed/merged please ?

**Pull request merged**